numpy/core/src/multiarray/mapping.c

```diff
@@ -2479,13 +2480,18 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     int i;
     NPY_BEGIN_THREADS_DEF;
 
-    if (mit->size == 0) {
-        /* All indices got broadcast away, do *not* check as it always was */
+    intp_type = PyArray_DescrFromType(NPY_INTP);
+
+    if (NpyIter_GetIterSize(mit->outer) == 0) {
+        /*
+         * There are two reasons why `mit->size` can be 0, either the
+         * advanced indices have no elements (after broadcasting) or the
+         * subspace is empty. When the subspace is empty, we will check
+         * whether the indices are correct, otherwise not.
+         */
```

Comment:
Is this comment still relevant now that `mit->size` is not used?

Reply:
Yeah, no... made a "suggestion" above...